armory/data/datasets.py

```diff
@@ -279,12 +279,44 @@ def _generator_from_tfds(
     return generator
 
 
+class MnistContext:
+    def __init__(self):
+        self.default_float = np.float32
+        self.quantization = 255
+        self.x_dimensions = (None, 28, 28, 1)
+
+
+mnist_context = MnistContext()
+
+
+def mnist_dataset_canonical_preprocessing(batch):
+    if batch.ndim != len(mnist_context.x_dimensions):
+        raise ValueError(
+            f"input batch dim {batch.ndim} != {len(mnist_context.x_dimensions)}"
+        )
+    for dim, (source, target) in enumerate(
+        zip(batch.shape, mnist_context.x_dimensions)
+    ):
+        pass
+    assert batch.dtype == np.uint8
+    assert batch.shape[2:] == mnist_context.x_dimensions[2:]
```

Comment:
should this be 
```suggestion
    assert batch.shape[1:] == mnist_context.x_dimensions[1:]
```

Reply:
Good catch! Updated the same bug on CIFAR as well.